ovs/lib/albanodecontroller.py

```diff
@@ -142,59 +149,137 @@ def remove_disk(alba_backend_guid, node_guid, disk, expected_safety):
-                raise RuntimeError('Cannot remove disk {0} as the current safety is not as expected ({1} vs {2})'.format(disk, final_safety, expected_safety))
+                raise RuntimeError('Cannot remove ASD {0} as the current safety is not as expected ({1} vs {2})'.format(asd_id, final_safety, expected_safety))
 
-            AlbaController.remove_units(alba_backend_guid, [disk_asd_id])
-            result = node.client.remove_disk(disk)
+            AlbaController.remove_units(alba_backend.guid, [asd_id])
+            result = node.client.delete_asd(disk_id, asd_id)
             if result['_success'] is False:
-                raise RuntimeError('Error removing disk: {0}'.format(result['_error']))
+                raise RuntimeError('Error removing ASD: {0}'.format(result['_error']))
         else:
             # Forcefully remove disk -> decommission-osd
             # no safety checks, don't call HTTP client
-            logger.warning('Alba decommission osd {0}'.format(disk_asd_id))
-            AlbaController.remove_units(alba_backend_guid, [disk_asd_id], absorb_exception=True)
+            logger.warning('Alba decommission osd {0} without safety validations (node down)'.format(asd_id))
+            AlbaController.remove_units(alba_backend.guid, [asd_id], absorb_exception=True)
 
-        if len(asds) == 1:
-            asds[0].delete()
+        asd.delete()
         node.invalidate_dynamics()
         alba_backend.invalidate_dynamics()
         alba_backend.backend.invalidate_dynamics()
         if node.storagerouter is not None:
             DiskController.sync_with_reality(node.storagerouter_guid)
-        return True
+
+        return disk_id
+
+    @staticmethod
+    @celery.task(name='albanode.reset_asd')
+    def reset_asd(node_guid, asd_id, expected_safety):
+        """
+        Removes and readds an ASD to a Disk
+        """
+        node = AlbaNode(node_guid)
+        disk_id = AlbaNodeController.remove_asd(node_guid, asd_id, expected_safety)
+        result = node.client.add_asd(disk_id)
+        if result['_success'] is False:
+            logger.error('Error resetting ASD: {0}'.format(result['_error']))
+            raise RuntimeError(result['_error'])
+
+    @staticmethod
+    @celery.task(name='albanode.restart_asd')
+    def restart_asd(node_guid, asd_id):
+        """
+        Restarts an ASD on a given Node
+        :param node_guid: Guid of the Node on which the ASD should be restarted
+        :param asd_id: ID of the ASD to be restarted
+        """
+        node = AlbaNode(node_guid)
+
+        try:
+            asds = node.client.get_asds()
+        except requests.ConnectionError:
+            logger.error('Alba HTTP Client connection failed on node {0}'.format(node.ip))
+            raise RuntimeError('Alba HTTP Client connection failed on node {0}'.format(node.ip))
+
+        disk_id = None
```

Comment:
This piece of code is used more often and could be a function or dynamic property